pallets/pool-system/src/lib.rs

```diff
@@ -1164,7 +1206,13 @@ pub mod pallet {
 			pool.execute_previous_epoch()?;
 
 			let executed_amounts = epoch.tranches.fulfillment_cash_flows(solution)?;
-			let total_assets = pool.reserve.total.ensure_add(epoch.nav)?;
+			// TODO: Check whether pool.reserve.total == epoch.reserve
+			// its not the same
+			let total_assets = pool
+				.reserve
+				.total
+				.ensure_add(epoch.nav)?
+				.ensure_sub(epoch.reserve)?;
 			let tranche_ratios = epoch.tranches.combine_with_residual_top(
```

Comment:
That's what I took from [your comment on DAO Slack ](https://centrifugedao.slack.com/archives/C04JJEBTDRN/p1704360815148039?thread_ts=1704358602.397019&cid=C04JJEBTDRN)or did I misunderstand something here?

Reply:
In unit tests, it became aware that `epoch.reserve` is not necessarily the same as `pool.reserve.total` at this codepath. So I am not sure whether this change is absolutely correct. Since none of the tests broke, I stuck with it.

Reply:
> the above should just be
> 
> ```rust
> let total_assets = epoch.nav;
> ```

This breaks many unit tests because during execution, `pool.reserve.total` is often much larger than the `pool.reserve.total` during epoch closing because of `pool.reserve.deposit_from_epoch(&epoch.tranches, solution)?;`.

The `total_assets` during execution include `pool.reserve.total` [currently on main branch](https://github.com/centrifuge/centrifuge-chain/blob/f101348b2255fa77f214c13528df87e425abde30/pallets/pool-system/src/lib.rs#L1167C4-L1167C65):
```
let total_assets = pool.reserve.total.ensure_add(epoch.nav)?;
```

So I feel like it is correct to change it to
```
let total_assets = pool.reserve.total.ensure_add(epoch.nav)?.ensure_sub(epoch.reserve)?;
```

If we then take `epoch.nav.ensure_sub(epoch.reserve)?` as NAV for tranche rebalancing, both the ratios as well the rebalncing uses `total_assets`, because in the rebalancing, the input NAV is increased by the `pool.reserve.total`: https://github.com/centrifuge/centrifuge-chain/blob/6535d758e74b648124297d75cfdfa36f518c5987/pallets/pool-system/src/tranches.rs#L986

Reply:
My above proposal would basically only reduce the `total_assets` of the new epoch `i+1` by the `PoolFeesNAV` of epoch `i` during epoch execution `i -> i+1`. Not sure if this is sufficient.

Reply:
426f4528e186ee93c1c166458d5441b0ca8a10c2

Reply:
Closed with #1695